app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/metrics_capture.rb

```diff
@@ -0,0 +1,112 @@
+    instance_id = metrics_endpoint.options["monitoring_instance_id"] unless metrics_endpoint.nil?
+    token = "Bearer " + IBMCloudSdkCore::IAMTokenManager.new(:apikey => ext_management_system.authentication_key("default")).access_token
+
+    response = RestClient::Request.execute(
+      :method  => :post,
+      :url     => "https://#{ext_management_system.provider_region}.monitoring.cloud.ibm.com/api/data",
+      :headers => {
+        'Content-Type'  => 'application/json',
+        'Authorization' => token,
+        'IBMInstanceID' => instance_id
+      },
+      :payload => File.open("metrics.json", 'rb')
+    )
+    data = JSON.parse(response.body)
+    dataset = consolidate_data(data["data"])
+
+    store_datapoints!(dataset[:timestamps], dataset[:cpu_usage_rate_average], "cpu_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:mem_usage_absolute_average], "mem_usage_absolute_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:net_usage_rate_average], "net_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:disk_usage_rate_average], "disk_usage_rate_average", counter_values_by_mor[target.ems_ref])
+
+    return counters_by_mor, counter_values_by_mor
+  rescue RestClient::ExceptionWithResponse => err
+    log_header = "[#{interval_name}] for: [#{target.class.name}], [#{target.id}], [#{target.name}]"
+    _log.error("#{log_header} Unhandled exception during perf data collection: [#{err}], class: [#{err.class}]")
+    _log.log_backtrace(err)
+    raise
+  end
+
+  def initialize_metrics_query(instance_name)
+    metrics_query = JSON.parse(File.read("metrics.json"))
+    metrics_query["filter"] = "ibm_resource_name = '#{instance_name}'"
+    File.write("metrics.json", JSON.pretty_generate(metrics_query))
+  end
+
+  def store_datapoints!(timestamps, datapoints, counter_key, counter_values_by_mor)
+    timestamps.zip(datapoints).each do |timestamp, datapoint|
+      counter_values_by_mor.store_path(Time.at(timestamp).utc, counter_key, datapoint)
+    end
+  end
+
+  def bps_to_kbps(datapoints)
+    datapoints.map! { |value| value / 1.kilobyte }
+  end
+
+  def consolidate_data(datapoints)
+    dataset = {:timestamps => [], :cpu_usage_rate_average => [], :mem_usage_absolute_average => [], :net_usage_rate_average => [], :disk_usage_rate_average => []}
+    datapoints.each do |datapoint|
+      dataset[:timestamps] << datapoint["t"]
+      dataset[:cpu_usage_rate_average] << datapoint["d"][0]
+      dataset[:mem_usage_absolute_average] << datapoint["d"][1]
+      dataset[:net_usage_rate_average] << datapoint["d"][2] + datapoint["d"][3]
+      dataset[:disk_usage_rate_average] << datapoint["d"][4] + datapoint["d"][5]
+    end
+    bps_to_kbps(dataset[:net_usage_rate_average])
+    bps_to_kbps(dataset[:disk_usage_rate_average])
```

Comment:
I think the data returned here is in bps since it is being aggregated over time as an average... but not entirely sure. Would be nice to get some eyes on https://docs.sysdig.com/en/docs/sysdig-monitor/metrics/data-aggregation/#time-aggregation